File: aurora-scala/src/main/scala/net/gree/aurora/scala/domain/shardingconfig/ShardingConfigIdImpl.scala
package net.gree.aurora.scala.domain.shardingconfig

import net.gree.aurora.domain.shardingconfig.{ShardingConfigId => JShardingConfigId}

private[domain]
final class ShardingConfigIdImpl(val underlying: JShardingConfigId) extends ShardingConfigId {

  def value: String = underlying.getValue

}

File: aurora-core/src/main/java/net/gree/aurora/domain/shardingconfig/ShardingConfigIdImpl.java
package net.gree.aurora.domain.shardingconfig;

import org.sisioh.dddbase.model.impl.AbstractIdentity;

final class ShardingConfigIdImpl extends AbstractIdentity<String> implements ShardingConfigId {

    /**
     * インスタンスを生成する。
     *
     * @param value 識別子の値
     */
    protected ShardingConfigIdImpl(String value) {
        super(value);
    }

}
